pkg/container/types/types.go

```diff
@@ -106,17 +107,78 @@ type Type struct {
 	Scale int32
 }
 
+// ProtoSize is used by gogoproto.
+func (t *Type) ProtoSize() int {
+	return 2*4 + 4*3
+}
+
+// MarshalToSizedBuffer is used by gogoproto.
+func (t *Type) MarshalToSizedBuffer(data []byte) (int, error) {
```

Comment:
what is the reason to marshal/unmarsh this thing?   Why dont we just return the 20 bytes as is?   Do we really care endianness?

Reply:
> This has a big problem -- we cannot depend anything on gogoproto.

But our project takes gogoproto as the default plugin of PB, see [proto/gen.sh](https://github.com/matrixorigin/matrixone/blob/352fa7db1643661ea83671af2088fda57a26418d/proto/gen.sh#L108).

`types.Type` is also the filed type of many other structures, so `MarshalToSizedBuffer` and other methods are implemented in order to facilitate the encoding/decoding for those structures.

This PR is mainly aimed at `de-gob` which replaces `gob` encoding with PB or manual encoding. By making use of PB, all we need to do is writing a `proto` file, then invoke logic from the generated `*.pb.go` with confidence, only if the `proto` file is proper or correct. All this could limit the risk for a larger PR.

According to [my test result](https://github.com/matrixorigin/matrixone/issues/7111#issuecomment-1518968009), this PR bring some improvement. Though that's not enough, but further optimization deserves another individual PR case by case.